packages/docusaurus-theme-classic/src/theme/DocSidebarItem/index.tsx

```diff
@@ -111,6 +114,28 @@ function DocSidebarItemCategory({
   });
 
   useAutoExpandActiveCategory({isActive, collapsed, setCollapsed});
+  const {expandedItem, setExpandedItem} = useDocSidebarItemsExpandedState();
+  function updateCollapsed(toCollapsed: boolean = !collapsed) {
+    setExpandedItem(toCollapsed ? null : index);
```

Comment:
The only case we have to think twice about UX is when there are multiple sidebar categories expanded by default. In this implementation, the `expandedItem` context is initialized to `null`, and when the user collapses one of those default-expanded categories, the others remain expanded until any category is toggled to be expanded.

Reply:
I’m excited this feature is finally getting some steam

Reply:
Yes! This is the UX I've been looking forward to, so I did invest some time in this :D

Reply:
I haven’t really looked at the code, just a quick glance but I would set the default setting to false, I know not everyone will want this feature on by default, and that was an issue when I first tried to get this in over a year ago 😂

Reply:
Yes, it's indeed defaulted to false :) Just that it would be cool to be turned on for our webste.